plugin-server/src/main/ingestion-queues/batch-processing/each-batch-async-handlers.ts

```diff
@@ -1,15 +1,27 @@
 import { EachBatchPayload, KafkaMessage } from 'kafkajs'
 
-import { RawClickHouseEvent } from '../../../types'
+import { PostIngestionEvent, RawClickHouseEvent } from '../../../types'
 import { convertToIngestionEvent } from '../../../utils/event'
 import { groupIntoBatches } from '../../../utils/utils'
 import { runInstrumentedFunction } from '../../utils'
 import { IngestionConsumer } from '../kafka-queue'
 import { eachBatch } from './each-batch'
 
 export async function eachMessageAsyncHandlers(message: KafkaMessage, queue: IngestionConsumer): Promise<void> {
-    const clickHouseEvent = JSON.parse(message.value!.toString()) as RawClickHouseEvent
-    const event = convertToIngestionEvent(clickHouseEvent)
+    // If the message isn't well formed JSON, just skip it. Ideally we'd send to a DLQ but I'll leave that as a TODO.
+    // TODO: Send to DLQ on message malformed
+    let clickHouseEvent: RawClickHouseEvent
+    let event: PostIngestionEvent
+    if (!message.value) {
+        return
+    } else {
+        try {
+            clickHouseEvent = JSON.parse(message.value.toString())
+            event = convertToIngestionEvent(clickHouseEvent)
+        } catch (error) {
+            return
```

Comment:
Shall we send it to Sentry for debugging?

Reply:
I'll add as a TODO as this was a slightly tangential thing to get the tests working. I was thinking DLQ rather than Sentry.

Reply:
And ensure that things that would have previously been send to Sentry still go there 👍

Reply:
Ok, I'm sending to Sentry now